index.bs

```diff
@@ -5638,6 +5638,93 @@ At this time, one [=credential property=] is defined: the [=resident key credent
+
+This extension is implemented on top of the [[FIDO-CTAP]] `hmac-secret` extension. It is a separate [=client extension=] because `hmac-secret` requires that inputs and outputs be encrypted in a manner that only the user agent can perform, and to provide domain separation between uses by WebAuthn and any uses by the underlying platform. The latter is achieved by hashing the provided PRF inputs with a context string to prevent evaluation of the PRFs for arbitrary inputs.
+
+Note: this extension may be implemented for [=authenticators=] that do not use [[FIDO-CTAP]] so long as the behavior observed by a [=[RP]=] is identical.
+
+: Extension identifier
+:: `prf`
+
+: Operation applicability
+:: [=registration extension|Registration=] and [=authentication extension|authentication=]
+
+: Client extension input ([=registration extension|registration=])
+::  <xmp class="idl">
+    partial dictionary AuthenticationExtensionsClientInputs {
+        boolean prfEnable;
+    };
+    </xmp>
+
+    <div dfn-type="dict-member" dfn-for="AuthenticationExtensionsClientInputs">
+        :   <dfn>prfEnable</dfn>
+        ::  If true, specifies that two PRFs should be associated with the created credential.
+    </div>
+
+: Client extension processing ([=registration extension|registration=])
+::
+       1. If {{AuthenticationExtensionsClientInputs/prfEnable}} is present and has the value `true`:
+           1. Set `hmac-secret` to `true` in the authenticator extensions input.
+           1. Set {{AuthenticationExtensionsClientOutputs/prfEnabled}} to the value of `hmac-secret` in the authenticator extensions output. If not present, set {{AuthenticationExtensionsClientOutputs/prfEnabled}} to false.
+       1. Else:
+           1. Set {{AuthenticationExtensionsClientOutputs/prfEnabled}} to false.
+
+: Client extension output ([=registration extension|registration=])
+:: <xmp class="idl">
+    partial dictionary AuthenticationExtensionsClientOutputs {
+        boolean prfEnabled;
+    };
+    </xmp>
+
+    <div dfn-type="dict-member" dfn-for="AuthenticationExtensionsClientOutputs">
+        :   <dfn>prfEnabled</dfn>
+        ::  True iff the two PRFs are available for use with the created credential.
+    </div>
+
+Note: it is not possible to evaluate a PRF during [=registration=] due to the design of the [[FIDO-CTAP]] `hmac-secret` extension.
+
+: Client extension input ([=authentication extension|authentication=])
+::  <xmp class="idl">
+    partial dictionary AuthenticationExtensionsClientInputs {
+        sequence<DOMString> prfEval;
+    };
+    </xmp>
+
+    <div dfn-type="dict-member" dfn-for="AuthenticationExtensionsClientInputs">
+        :   <dfn>prfEnable</dfn>
+        ::  A sequence of one or two strings that are inputs on which to evaluate the PRF.
```

Comment:
How about encoding that in the type? This is also how CTAP does it in `hmac-secret`.

```
    partial dictionary AuthenticationExtensionsClientInputs {
        required DOMString prfEval1;
        DOMString prfEval2;
    };
```

Reply:
Done.